luarules/gadgets/team_transfer/game_disable_assist_ally_construction.lua

```diff
@@ -0,0 +1,36 @@
+local API = VFS.Include("luarules/gadgets/team_transfer/api_gadgets.lua")
+local sharingModeUtils = VFS.Include("common/sharing_mode_utils.lua")
+local KEYS = VFS.Include("common/sharing_modoption_keys.lua")
+local Definitions = VFS.Include("luarules/gadgets/team_transfer/definitions.lua")
+
+if not sharingModeUtils.shouldGadgetRun(KEYS.DISABLE_ASSIST_ALLY_CONSTRUCTION) then
+	return
+end
+
+local allowAssist = not Spring.GetModOptions()[KEYS.DISABLE_ASSIST_ALLY_CONSTRUCTION]
+if allowAssist then
+	return
+end
+
+API.RegisterPolicy(function(policy)
+	policy:For(Definitions.PolicyType.Command)
+	:Use(function(ctx)
```

Comment:
Ok getting there, is this canonical LUA? `:Use(function(policy_ctx)` just hurts my brain for some reason I'm sure I'll get used to it. I like this, but I almost want this factored further. For example what is the following block DOING? It is ```lua
policy:For(Commands)
`:Then(function(commandCtx) {
   commandCtx:WhenGuarding():Reject():WhenAreAllied():WhenCanAssist()
    commandCtx:WhenRepairing():Reject:WhenAreAllied():TargetComplete()
 }```
 
 And to me this implies a similar domain factoring for outcomes/units|resources.lua (RejectUnitTransfer, RejectResourceTransfer behavior), triggers/units|resources.lua (when statements above)